### DISCUSSION TITLE
Realtime subscriptions removed on system sleep

### DISCUSSION THREAD

**Discussion by shankar-narayanan-d** [supabase/supabase#13046] [Questions]
I am using supabase to get realtime updates. If my client is active, I get the updates properly. But if my client goes to sleep mode, my subscriptions are cleaned up. Once the client is back it takes about two minutes to get the previous subscription back. Any updates between these two minutes is not recieved.
Sometimes, after my client is back from sleep mode, it returns TIMED_OUT message so my subscription is not back.

**Comment by GaryAustin1** [comment]
See this issue.
https://github.com/supabase/realtime-js/issues/121
You need to handle reconnecting and reloading any missing data when realtime drops connection (which it does for several cases).  Although it tries to reconnect when it can, you will still lose any data changes while the connection is gone.
